### PR TITLE
Feature/padatious

### PR DESCRIPTION
**Pull request by Mn0491**
This PR is to replace the old alarm skill with new and revamp one. Features include

- set alarm
- set a recurring alarm
- set an alarm using weekend and weekday
- delete an alarm
- status of an alarm
- converse 
- padatious
- event_scheduler
- alarm beeps can be turned off via mark 1 top button

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -15,95 +15,406 @@
...
+        self.register_intent_file('stop.intent', self.stop)
+        if self.time_format == 'half':
+            self.register_intent_file('set.time.intent', self.handle_set_time)
+            self.register_entity_file('ampm.entity')
+        elif self.time_format == 'full':
+            self.register_intent_file('set.time.24hr.intent',
+                                      self.handle_set_time)
+        self.register_entity_file('time.entity')
+        self.register_entity_file('length.entity')
+        self.register_entity_file('daytype.entity')
```
Comment (MatthewScholefield) [inline __init__.py]: We should change all these to the decorator form.

File: __init__.py [hunk 2]
```diff
@@ -15,95 +15,406 @@
...
-        with self.file_system.open(self.PENDING_TASK, 'w') as f:
-            yaml.safe_dump(self.data, f)
-        with self.file_system.open(self.REPEAT_TASK, 'w') as f:
-            yaml.safe_dump(self.repeat_data, f)
-        if not self.alarm_on:
-            self.schedule()
+        # self.register_intent_file('set.morning.intent', self.set_morning)
+        # self.register_intent_file('set.sunrise.intent', self.set_sunrise)
+        # self.register_intent_file('delete.all.intent', self.delete_all)
+        # self.register_entity_file('exceptdaytype.entity')
```
Comment (MatthewScholefield) [inline __init__.py]: Should we remove these and if these are to-be implemented, paste them into an issue?

File: __init__.py (hunk 1 above)
Comment (Mn0491) [reply __init__.py]: @MatthewScholefield what does the decorator form look like?

Comment (MatthewScholefield) [reply __init__.py]: [Here's the documentation](https://github.com/MycroftAI/mycroft-core/wiki/Intent-handler-decorators#using-intent-handler-decorators). For Padatious it's `@intent_file_handler(...`.

**Comment by Mn0491** [comment]
@MatthewScholefield changes made

File: __init__.py [hunk 3]
```diff
@@ -15,95 +15,447 @@
...
+            speak utterance
+
+            Args:
+                message_data (dict): a dictionary from the intent message bus
+        """
+        alarm_object = self.parse_message_data(message_data)
+
+        if alarm_object['length'] != "":
+            d = dparser.parse(alarm_object["name"])
+            now = datetime.now()
```
Comment (penrods) [inline __init__.py]: This gives the local time, but should be working in UTC.  You can use mycroft.util.parser.extract_datetime("") to get UTC now.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+git+git://github.com/crsmithdev/arrow.git@b33925e9ce55c3544afcb66a0ffb205593b49ab7
```
Comment (MatthewScholefield) [inline requirements.txt]: They updated so this shouldn't need a hash anymore

File: __init__.py (hunk 3 above)
Comment (Mn0491) [reply __init__.py]: fixed using 

```python
        user_set_tz = \
            timezone(self.time_zone).localize(datetime.now()).strftime('%Z')
        device_tz = time.tzname
        if user_set_tz in device_tz:
            return arrow.get(d)
        else:
            seconds_to_shift = int(self.location['timezone']['offset']) / -1000
            return arrow.get(d).shift(seconds=seconds_to_shift)
```

**Comment by penrods** [comment]
Merging for now, definitely better than it was!

**Pull request merged**